File: LEETCODE/9. 回文数.java
🔗https://leetcode-cn.com/problems/palindrome-number/

class Solution {
    public boolean isPalindrome(int x) {
        if(x<0)
            return false;
        int num1 = x;
        int num2 = 0;
        while(x != 0){
            if(x%10==0)
                num2 = num2*10;
            else
                num2 = num2*10 + x%10; 
            x /= 10;
        }
        if(num1==num2)
            return true;
        else
            return false;
    }
}

File: LEETCODE/kotlin/9. 回文数.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x<0) return false
        var x1 = x
        var x2 = 0
        while (x1!=0) {
            x2 = if(x1%10==0) x2*10 else x2*10+x1%10
            x1 /= 10
        }
        return if(x2==x) true else false;
    }
}